### PR TITLE
Release doc

### PR DESCRIPTION
**Pull request by brybacki**
This is based of the cdi release doc
https://github.com/kubevirt/containerized-data-importer/blob/main/doc/releases.md.

Signed-off-by: Bartosz Rybacki <brybacki@redhat.com>

**What this PR does / why we need it**:

Documents steps needed to release a plugin.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/build/releases.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- Patch - mid-Sprint release for fixing blocker bugs. In the case that a bug is blocking kubevirt-velero-plugin consumers' workflow, 
+a fix may be released as soon as it is merged.  A Patch should be limited expressly to the bug fix and not include anything unrelated.
+
+## Releasing a New Version
+
+Release branches are used to isolate a stable version of kubevirt-velero-plugin.  Git tags are used within these 
+release branches to track incrementing of Minor and Patch versions.  When a Major version is incremented, a new 
+stable branch should be created corresponding to the release.
+
+- Release branches should adhere to the `release-v#.#.#` pattern.
```
Comment (awels) [inline hack/build/releases.md:34]: release branches are major/minor, so X.Y, not X.Y.Z, Zs are tags.

Comment (brybacki) [reply hack/build/releases.md]: done

**Pull request merged**